graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -208,18 +240,39 @@ public GraphqlService build() {
         return new DefaultGraphqlService(builder.build(), dataLoaderRegistry, useBlockingTaskExecutor);
     }
 
-    private static TypeDefinitionRegistry typeDefinitionRegistry(List<File> schemaFiles) {
+    private GraphQLSchema makeSchema() {
+        if (schema != null) {
```

Comment:
If the `schema` is not used with `schemaURLs`, `runtimeWiringConfigurators`, and `typeVisitors `, how about checking the state when the `schema` is set and vice-versa?
`checkState(schemaURLs == null && runtimeWiringConfigurators == null && typeVisitors == null, ...)`
(We need to create `ImmutableList.builder()` when the parameters are set by the way.)

Reply:
How about logging a warning rather than using `checkState(...)` ?

Reply:
If a user can catch an exception directly, it would be kind to raise an exception.

Reply:
Well, we can leave a warning but I slightly prefer to raise an exception so that users can easily notice the mutual exclusive condition when the service is built.
Also, we conventionally throw an `IllegalStateException`. e.g. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java#L275